optimum/habana/transformers/trainer.py

```diff
@@ -375,25 +380,8 @@ def _wrap_model(self, model, training=True, dataloader=None):
             return model
 
         if self.args.local_rank != -1:
-            kwargs = {}
-
-            kwargs["find_unused_parameters"] = self.args.ddp_find_unused_parameters
-            if self.args.ddp_find_unused_parameters and self.args.gradient_checkpointing:
-                logger.warning(
-                    "ddp_find_unused_parameters and gradient_checkpointing are both True, which may lead to an error:"
-                    " https://github.com/huggingface/transformers/pull/4659#issuecomment-643356021"
-                )
-            kwargs["bucket_cap_mb"] = self.args.ddp_bucket_cap_mb
-
-            if self.args.use_habana:
-                kwargs["gradient_as_bucket_view"] = True
-
-            model = torch.nn.parallel.DistributedDataParallel(
-                model,
-                device_ids=[self.args.local_rank] if self.args._n_gpu != 0 and not self.args.use_habana else None,
-                output_device=self.args.local_rank if self.args._n_gpu != 0 and not self.args.use_habana else None,
-                **kwargs,
-            )
+            from .fastddp import FastDistributedDataParallel
+            model = FastDistributedDataParallel(model, fusion_buffer_dtype=torch.bfloat16)
```

Comment:
Linked to one question above, does `FastDistributedDataParallel` aim at completely replacing `torch.nn.parallel.DistributedDataParallel`? Or is it just for HPU graphs?

Reply:
It completely replaces DDP to minimize complexity. If you don't like it, we can proceed without FastDDP, yet the scaling factor will be <90% (my goal was to improve the scaling factor).